### PR TITLE
Support 'float' and 'java.lang.Float'

### PR DESCRIPTION
**Pull request by yannmh**
## Summary

Support `float` and `java.lang.Float` attribute types as simple JMX attributes. Assert the support of these attribute types with tests.

### Support `float` and `java.lang.Float`
Support `float` and `java.lang.Float` attribute types as simple JMX attributes. 

### `TestCommon` to refactor JUnit tests
Introduce a `TestCommon` parent class to refactor JUnit tests.
Features:
* One-liner `registerMBean` method
* One-liner `initApplication` nethod
* Automatically unregister MBeans after execution of every test
* More eye candies

### `assertMetric` and `assertCoverage` methods
Add two new methods to `TestCommon` test class to refactor JUnit tests.
* `assertMetric`: assert that a specific metric was collected
* `assertCoverage`: assert that all -excluding JVM related- metrics were
tested.

Refactor `TestApp` tests.

### Attribute types coverage
Assert the support of these attribute types with tests.

### CODE REVIEW

File: src/test/java/org/datadog/jmxfetch/TestApp.java [hunk 1]
```diff
@@ -64,259 +54,180 @@ public void testBeanTags() throws Exception {
...
-        app.doIteration();
-        LinkedList<HashMap<String, Object>> metrics = ((ConsoleReporter) appConfig.getReporter()).getMetrics();
-
-        // First filter 26 = 13 metrics from java.lang + 13 metrics implicitly defined
-        assertEquals(26, metrics.size());
+        run();
+        LinkedList<HashMap<String, Object>> metrics = getMetrics();
 
-        mbs.unregisterMBean(includeObjectName);
+        // First filter 27 = 13 metrics from java.lang + 14 metrics implicitly defined
```
Comment (olivielpeau) [inline src/test/java/org/datadog/jmxfetch/TestApp.java]: 28 = 13 + 15 :wink:

File: src/test/java/org/datadog/jmxfetch/TestApp.java [hunk 2]
```diff
@@ -332,397 +243,104 @@ public void testExitWatcher() throws Exception {
...
 
-        app.doIteration();
-        ConsoleReporter reporter = ((ConsoleReporter) appConfig.getReporter());
+        run();
+        LinkedList<HashMap<String, Object>> metrics = getMetrics();
 
-        LinkedList<HashMap<String, Object>> metrics = reporter.getMetrics();
-
-        assertEquals(25, metrics.size()); // 25 = 13 metrics from java.lang + the 5 gauges we are explicitly collecting + the 7 gauges that is implicitly collected, see jmx.yaml in the test/resources folder
+        assertEquals(27, metrics.size()); // 26 = 13 metrics from java.lang + the 5 gauges we are explicitly collecting + the 9 gauges that is implicitly collected, see jmx.yaml in the test/resources folder
```
Comment (olivielpeau) [inline src/test/java/org/datadog/jmxfetch/TestApp.java]: `27` in the comment :pig_nose:

File: src/test/java/org/datadog/jmxfetch/TestApp.java [hunk 3]
```diff
@@ -332,397 +243,104 @@ public void testExitWatcher() throws Exception {
...
+        assertMetric("jmx.org.datadog.jmxfetch.test.instance_float", 567.8f, commonTags, 5);
+        assertCoverage();
 
         // We run a second collection. The counter should now be present
-        app.doIteration();
-        metrics = ((ConsoleReporter) appConfig.getReporter()).getMetrics();
-        assertEquals(27, metrics.size()); // 27 = 13 metrics from java.lang + the 5 gauges we are explicitly collecting + 7 gauges implicitly collected + 2 counter, see jmx.yaml in the test/resources folder
+        run();
+        metrics = getMetrics();
+        assertEquals(29, metrics.size()); // 28 = 13 metrics from java.lang + the 5 gauges we are explicitly collecting + 9 gauges implicitly collected + 2 counter, see jmx.yaml in the test/resources folder
```
Comment (olivielpeau) [inline src/test/java/org/datadog/jmxfetch/TestApp.java]: `29` in the comment :eyes:

File: src/test/java/org/datadog/jmxfetch/TestCommon.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+import org.junit.After;
+import org.junit.BeforeClass;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import com.beust.jcommander.JCommander;
+import org.apache.log4j.Level;
+
+
+public class TestCommon {
```
Comment (olivielpeau) [inline src/test/java/org/datadog/jmxfetch/TestCommon.java]: Could you check the indentation in this file? It seems to be using mostly tabs, but we're using spaces in all the other files.

**Comment by olivielpeau** [comment]
Love the refactoring of the tests :heart:

Looks great, just tiny comments/indentation to fix, apart from that :shipit:

**Comment by yannmh** [comment]
Thanks for the review @olivielpeau :1234: :heavy_plus_sign: :1234:

**Pull request merged**